### PR TITLE
improve required param error handling

### PR DESCRIPTION
**Pull request by zandre-eng**
Certain input parameters are required for some of the workflows, such as the template string for verification. This PR adds in additional checks and error handling to stop the users from completing a certain workflow if there are missing required input parameters.

### CODE REVIEW

File: app/src/main/java/com/dimagi/biometric/activities/BaseActivity.java [hunk 1]
```diff
@@ -71,6 +71,13 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     protected void loadFragment(Fragment fragment) {
+        ArrayList<String> errors = validateRequiredParams();
+        if (errors.size() > 0) {
+            Bundle args = new Bundle();
+            String errorStr = createErrorStr(errors);
+            args.putString("errors", errorStr);
+            fragment.setArguments(args);
+        }
```
Comment (avazirna) [inline app/src/main/java/com/dimagi/biometric/activities/BaseActivity.java]: Can we refactor this to a separate method?

File: app/src/main/java/com/dimagi/biometric/activities/VerifyActivity.java [hunk 2]
```diff
@@ -33,4 +33,16 @@ protected void onCaptureCancelled() {
...
         ).finalizeResponse(this);
     }
+
+    @Override
+    protected ArrayList<String> validateRequiredParams() {
+        ArrayList<String> errors = new ArrayList<>();
+        if (caseId == null) {
+            errors.add(getText(R.string.missing_case_id).toString());
+        }
+        if (templateStr == null) {
```
Comment (avazirna) [inline app/src/main/java/com/dimagi/biometric/activities/VerifyActivity.java:43]: I don't see where `templateStr` is declared. Did I miss it?

File: app/src/main/java/com/dimagi/biometric/fragments/BaseMatchFragment.java [hunk 3]
```diff
@@ -71,6 +70,23 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
     }
 
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        Bundle args = getArguments();
+        String errors = null;
+        if (args != null) {
+            errors = args.getString("errors");
+        }
```
Comment (avazirna) [inline app/src/main/java/com/dimagi/biometric/fragments/BaseMatchFragment.java]: Same here, can we refactor this

**Review by avazirna — COMMENTED** [review]
@zandre-eng Overall, looks good, I just have a few minor comments

File: app/src/main/java/com/dimagi/biometric/activities/VerifyActivity.java (hunk 2 above)
Comment (zandre-eng) [reply app/src/main/java/com/dimagi/biometric/activities/VerifyActivity.java:43]: Ah, I seem to have by mistake put the declaration of this variable in a [later PR](https://github.com/dimagi/biometric-android/pull/6). These should all be merged together, so this should be fine.

File: app/src/main/java/com/dimagi/biometric/activities/BaseActivity.java (hunk 1 above)
Comment (zandre-eng) [reply app/src/main/java/com/dimagi/biometric/activities/BaseActivity.java]: [71a915d](https://github.com/dimagi/biometric-android/pull/3/commits/71a915d5adbdedb8a00e085a5796b687e9903e21)

File: app/src/main/java/com/dimagi/biometric/fragments/BaseMatchFragment.java (hunk 3 above)
Comment (zandre-eng) [reply app/src/main/java/com/dimagi/biometric/fragments/BaseMatchFragment.java]: [71a915d](https://github.com/dimagi/biometric-android/pull/3/commits/71a915d5adbdedb8a00e085a5796b687e9903e21)

**Review by avazirna — APPROVED** [review]

**Pull request merged**